ui/apps/platform/src/reducers/metadata.ts

```diff
@@ -0,0 +1,112 @@
+import { Reducer, combineReducers } from 'redux';
+import isEqual from 'lodash/isEqual';
+
+import { Metadata } from 'types/metadataService.proto';
+import { PrefixedAction } from 'utils/fetchingReduxRoutines';
+
+// Action types
+
+type InitialFetchMetadataAction = PrefixedAction<'metadata/INITIAL_FETCH_METADATA', Metadata>;
+type PollMetadataAction = PrefixedAction<'metadata/POLL_METADATA', Metadata>;
+
+type MetadataAction = InitialFetchMetadataAction | PollMetadataAction;
+
+// Action creator functions
+
+// TODO Replaced abstract with concrete for compatibility with sagas, but simplify to action objects for thunks, if possible.
+export const actions = {
+    initialFetchMetadata: {
+        failure: (error: Error) => ({
+            type: 'metadata/INITIAL_FETCH_METADATA_FAILURE',
+            error,
+        }),
+        success: (response: Metadata) => ({
+            type: 'metadata/INITIAL_FETCH_METADATA_SUCCESS',
+            response,
+        }),
+    },
+    pollMetadata: {
+        failure: (error: Error) => ({
+            type: 'metadata/POLL_METADATA_FAILURE',
+            error,
+        }),
+        success: (response: Metadata) => ({
+            type: 'metadata/POLL_METADATA_SUCCESS',
+            response,
+        }),
+    },
+};
+
+// Reducers
+
+// Initial state arbitrarily assumes release build.
+const metadataInitialState: Metadata = {
+    buildFlavor: 'release',
+    licenseStatus: 'VALID',
+    releaseBuild: true,
+    version: '', // response for request before authentication does not reveal version
+};
+
+const metadata: Reducer<Metadata, MetadataAction> = (
+    state = metadataInitialState,
+    action
+): Metadata => {
+    switch (action.type) {
```

Comment:
The benefits of typing this file out can also be noticed when highlighting properties such as `type` here

![Screenshot 2023-02-21 at 11 16 28 AM](https://user-images.githubusercontent.com/61400697/220414601-9bea702e-6e49-4438-980d-3444fabc433e.png)